### PR TITLE
Issue/4296 hardcoded strings

### LINKED ISSUE #4296: Extract hard-coded strings in Tests
Most of the time values are being matched with hard-coded strings, while we have the `words` method where all printed values are being stored. It would make sense to use the `words` instead. It makes maintenance easier. When some texts in the UI change, like casings being aligned, we wouldn't have to hunt down all places where that string was hardcoded in tests.

### PR DESCRIPTION
**Pull request by matborowczyk**
# Description

* Short description here *

closes #4296 

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [x] Attached issue to pull request
- [x] Changelog entry
- [x] Code is clear and sufficiently documented
~~- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)~~
- [x] Correct, in line with design
~~- [ ] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )~~

# Reviewer Checklist:

- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [ ] Code is clear and sufficiently documented
- [ ] Correct, in line with design

### CODE REVIEW

File: src/Slices/CompileReports/UI/Page.test.tsx [hunk 1]
```diff
@@ -228,12 +231,17 @@ test("When using the status filter with the In Progress opiton then the compile
...
+    screen.getByRole("option", { name: words("compileReports.columns.status") })
+  );
 
-  const input = screen.getByPlaceholderText("Select compile status...");
+  const input = screen.getByPlaceholderText(
+    words("compileReports.filters.status.placeholder")
+  );
   await userEvent.click(input);
 
   const option = await screen.findByRole("option", { name: "in progress" });
```
Comment (LauraStordeur) [inline src/Slices/CompileReports/UI/Page.test.tsx:243]: Did you not extract the "in progress" because it isn't yet available in the "words" dict?

Comment (LauraStordeur) [reply src/Slices/CompileReports/UI/Page.test.tsx:243]: The idea was to extract all hardcoded strings, that are displayed in the UI. "success", "in progress" are hardcoded in your test here, and are values that are displayed to the UI. Doesn't matter where the string comes from, you should do the assertion/check/search based on the 'words' dict, not a magic string :)

File: src/Slices/Home/UI/EnvironmentsOverview.test.tsx [hunk 2]
```diff
@@ -49,7 +50,9 @@ test("Given environments overview When filtering by project Then only the matchi
     name: "Environment card",
   });
   expect(initialCards).toHaveLength(4);
-  const input = await screen.findByPlaceholderText("Filter by project");
+  const input = await screen.findByPlaceholderText(
+    words("home.filters.project.placeholder")
+  );
   await userEvent.click(input);
   const option = await screen.findByRole("option", { name: "default" });
```
Comment (LauraStordeur) [inline src/Slices/Home/UI/EnvironmentsOverview.test.tsx:57]: if default is printed to the screen, this has to be extracted too.

File: src/Slices/Home/UI/EnvironmentsOverview.test.tsx [hunk 3]
```diff
@@ -67,11 +70,15 @@ test("Given environments overview When filtering by name and project Then only t
     name: "Environment card",
   });
   expect(initialCards).toHaveLength(4);
-  const projectInput = await screen.findByPlaceholderText("Filter by project");
+  const projectInput = await screen.findByPlaceholderText(
+    words("home.filters.project.placeholder")
+  );
   await userEvent.click(projectInput);
   const option = await screen.findByRole("option", { name: "default" });
```
Comment (LauraStordeur) [inline src/Slices/Home/UI/EnvironmentsOverview.test.tsx:77]: same here

File: src/Slices/Parameters/UI/Page.test.tsx [hunk 4]
```diff
@@ -152,7 +159,11 @@ test("When using the Updated filter then the parameters within the range selecte
...
     )
   );
-  await userEvent.click(screen.getByRole("option", { name: "Updated" }));
+  await userEvent.click(
+    screen.getByRole("option", {
+      name: words("parameters.columns.updated.tests"),
+    })
+  );
 
   const fromDatePicker = screen.getByLabelText("From Date Picker");
```
Comment (LauraStordeur) [inline src/Slices/Parameters/UI/Page.test.tsx:168]: labels are printed to the UI

File: src/Slices/ServiceInventory/UI/Spec/Filter.spec.ts [hunk 5]
```diff
@@ -24,7 +25,9 @@ test("GIVEN The Service Inventory WHEN the user filters on something THEN a data
   });
   expect(beforeRows.length).toEqual(2);
 
-  const input = await screen.findByPlaceholderText("Select a state...");
+  const input = await screen.findByPlaceholderText(
+    words("inventory.filters.state.placeholder")
+  );
   await userEvent.click(input);
 
   const option = await screen.findByRole("option", { name: "creating" });
```
Comment (LauraStordeur) [inline src/Slices/ServiceInventory/UI/Spec/Filter.spec.ts:33]: option visible in UI?

File: src/Slices/Status/UI/Page.test.tsx [hunk 6]
```diff
@@ -163,9 +164,13 @@ test("GIVEN StatusPage with support extension WHEN user click download THEN an a
   const downloadButton = screen.getByRole("button", {
     name: "DownloadArchiveButton",
```
Comment (LauraStordeur) [inline src/Slices/Status/UI/Page.test.tsx:165]: why is it that some button names are extracted, and others aren't?

Comment (LauraStordeur) [reply src/Slices/Status/UI/Page.test.tsx:165]: Github prevents me from commenting on lines that weren't actively part of the changes, so I can't point out the ones you missed. But There are a few still I couldn't comment on...

File: src/Slices/CompileReports/UI/Page.test.tsx [hunk 7]
```diff
@@ -276,7 +286,11 @@ it("When using the Date filter then the compile reports within the range selecte
...
     )
   );
-  await userEvent.click(screen.getByRole("option", { name: "Requested" }));
+  await userEvent.click(
+    screen.getByRole("option", {
+      name: words("compileReports.columns.requested"),
+    })
+  );
 
   const fromDatePicker = await screen.findByLabelText("From Date Picker");
```
[the remark this answers is not in the document]
Comment (matborowczyk) [reply src/Slices/CompileReports/UI/Page.test.tsx:295]: That's a text for aria label,  it not shown anywhere

File: src/Slices/DesiredState/UI/Page.test.tsx [hunk 8]
```diff
@@ -290,7 +295,9 @@ test("When using the Version filter then the desired state versions within the r
       { name: "FilterPicker" }
     )
   );
-  await userEvent.click(screen.getByRole("option", { name: "Version" }));
+  await userEvent.click(
+    screen.getByRole("option", { name: words("desiredState.columns.version") })
+  );
 
   const fromDatePicker = await screen.findByLabelText("Version range from");
```
[the remark this answers is not in the document]
Comment (matborowczyk) [reply src/Slices/DesiredState/UI/Page.test.tsx:302]: as one comment above, it's aria label, not visible label

File: src/Slices/Home/UI/EnvironmentsOverview.test.tsx (hunk 2 above)
Comment (matborowczyk) [reply src/Slices/Home/UI/EnvironmentsOverview.test.tsx:57]: There is no such a text visible in home screen

File: src/Slices/Home/UI/EnvironmentsOverview.test.tsx (hunk 3 above)
Comment (matborowczyk) [reply src/Slices/Home/UI/EnvironmentsOverview.test.tsx:77]: same as above

File: src/Slices/Parameters/UI/Page.test.tsx (hunk 4 above)
Comment (matborowczyk) [reply src/Slices/Parameters/UI/Page.test.tsx:168]: This one isn't it's just aria-label

File: src/Slices/Status/UI/Page.test.tsx (hunk 6 above)
Comment (matborowczyk) [reply src/Slices/Status/UI/Page.test.tsx:165]: visible text for that button is "Download Archive Button", and only corresponding for value visible above is aria-label

answering second comment: I went through others and for them answer is similar, value asked in parameter "name" is corresponding to aria-label, not visible

File: src/Slices/ServiceInventory/UI/Spec/Filter.spec.ts (hunk 5 above)
Comment (matborowczyk) [reply src/Slices/ServiceInventory/UI/Spec/Filter.spec.ts:33]: it is,I'll change it

File: src/Slices/CompileReports/UI/Page.test.tsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (LauraStordeur) [reply src/Slices/CompileReports/UI/Page.test.tsx:297]: okay, damn that's confusing x)

File: src/Slices/Status/UI/Page.test.tsx (hunk 6 above)
Comment (LauraStordeur) [reply src/Slices/Status/UI/Page.test.tsx:165]: okay :D

**Review by LauraStordeur — APPROVED** [review]

**Pull request closed without merging**